### PR TITLE
Support for mutual TLS on the CTS API

### PR DESCRIPTION
**Pull request by mkam**
Adds support for enabling mutual TLS by setting `verify_incoming` to `true` and setting either `ca_cert` or `ca_path`. If both are provided the `ca_cert` will take precedence over `ca_path`, per the [hashicorp/go-rootcerts](https://github.com/hashicorp/go-rootcerts) library behavior.

Changes:
- Support for mutual TLS
- NewAPI can now return an error since the mTLS configs are loaded in this method
- Created a bunch of new test certificates and added a README to describe each one
- Moved all the certificates to their own directory, updated all the paths

Example Configuration:
```
tls {
  enabled = true
  cert = "/path/to/certs/localhost_cert2.pem"
  key = "/path/to/certs/localhost_key2.pem" 
  verify_incoming = true
  ca_cert = "/path/to/certs/localhost_cert.pem"
}
```
```
tls {
  enabled = true
  cert = "/path/to/localhost_cert2.pem"
  key = "/path/to/localhost_key2.pem" 
  verify_incoming = true
  ca_path = "/path/to/certs/"
}
```

TODO:
- Would like to print a warning message if both ca_path and ca_cert are provided.
- The code we're using for TLS logs errors in its own format. I'd like to figure out how to wrap this with our logging format:
```
2021-11-09T12:24:13.167-0600 [INFO]  api: received request: reqID=36c73044-5963-aa8c-8054-4a19fe979463 time=2021-11-09T12:24:13.167-0600 remote_ip=[::1]:54567 uri=/v1/status method=GET host=localhost:8558
2021-11-09T12:24:13.167-0600 [INFO]  api: request complete: reqID=36c73044-5963-aa8c-8054-4a19fe979463 duration=55us
2021/11/09 12:30:44 http: TLS handshake error from [::1]:54789: tls: failed to verify client certificate: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "localhost")
```

### CODE REVIEW

File: testutils/certs/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Test Certificates
+
+## Consul Certificate
+ `consul_cert.pem` is a self-signed certificate used for configuring Consul with TLS. The private key for the certificate is `consul_key.pem`. The same certificate is used by CTS as the CA authority for communicating with Consul on a TLS connection.
+
+The DNS name for the certificate is `example.com` and the IP address is `127.0.0.1`. The certificate has a key usage restriction where it cannot be used for client authentication.
+
+<details>
+<summary>Example Consul Configuration Snippet</summary>
+```
```
Comment (lornasong) [inline testutils/certs/README.md:10]: This nice markdown formatting isn't rendering as expected for me [here](https://github.com/hashicorp/consul-terraform-sync/blob/3f9b1adef0844c2970be48b0068ad009e9792674/testutils/certs/README.md)

Screenshot:
Screen Shot 2021-11-09 at 3 09 52 PM

File: testutils/certs/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Test Certificates
```
Comment (lornasong) [inline testutils/certs/README.md:1]: Thanks for writing up all these helpful details!

**Review by lornasong — APPROVED** [review]
looks good! commits were very readable and easy to understand. tests were helpful to understand how the the config and certs should be used

File: testutils/certs/README.md (hunk 1 above)
Comment (mkam) [reply testutils/certs/README.md:10]: Thanks for checking this! Looks like it needed a newline between the summary and the codeblock.

**Pull request merged**